packages/backend/src/clients/ResultsFileStorageClients/S3ResultsFileStorageClient.ts

```diff
@@ -98,11 +100,16 @@ export class S3ResultsFileStorageClient extends S3CacheClient {
     }
 
     async getFileUrl(cacheKey: string, fileExtension = 'jsonl') {
+        // Check if the cacheKey already has a file extension
+        const hasExtension = cacheKey.includes('.');
```

Comment:
Cache keys always have a `.` 
```
const queryHashKey = resultsIdentifiers.timezone
            ? `${projectUuid}.${resultsIdentifiers.sql}.${resultsIdentifiers.timezone}`
            : `${projectUuid}.${resultsIdentifiers.sql}`;
```

Should we check if the cacheKey ends with `.${fileExtension}`?